actionview/lib/action_view/helpers/tags/select.rb

```diff
@@ -37,7 +37,12 @@ def render
           #   [nil, []]
           #   { nil => [] }
           def grouped_choices?
-            !@choices.blank? && @choices.first.respond_to?(:second) && Array === @choices.first.second
+            return false if @choices.blank?
+
+            first_choice = @choices.first
+            return false unless first_choice.is_a?(Array) || first_choice.is_a?(Hash)
```

Comment:
Maybe we can check if it's an Enumerable? It seems that this was the intended behaviour according to https://github.com/rails/rails/commit/efe62b71d52c40849ead8855fb4f42fae91d4c54

Reply:
Good point.

Reply:
I think I'm wrong actually, I edited my comment.

Reply:
Would we want to allow grouped_choices for an Enumerable of Array?

I _think_ so?

Reply:
> Would we want to allow grouped_choices for an array of Enumerable?

Ah yes, that would make sense to me